components/picker/index.css

```diff
@@ -321,47 +337,28 @@ governing permissions and limitations under the License.
 		}
 	}
 
-	&:disabled,
-	&.is-disabled {
-		cursor: default;
-		background-color: var(--highcontrast-picker-background-color, var(--mod-picker-background-color-disabled, var(--spectrum-picker-background-color-disabled)));
-		border-color: var(--highcontrast-picker-border-color-disabled, transparent);
-		color: var(--highcontrast-picker-content-color-disabled, var(--mod-picker-font-color-disabled, var(--spectrum-picker-font-color-disabled)));
-
-		.spectrum-Picker-icon,
-		.spectrum-Picker-menuIcon,
-		.spectrum-Picker-validationIcon {
-			color: var(--highcontrast-picker-content-color-disabled, var(--mod-picker-icon-color-disabled, var(--spectrum-picker-icon-color-disabled)));
-		}
-
-		.spectrum-Picker-label.is-placeholder {
-			color: var(--highcontrast-picker-content-color-disabled, var(--mod-picker-font-color-disabled, var(--spectrum-picker-font-color-disabled)));
-		}
-	}
-
 	.spectrum-Picker-icon {
 		flex-shrink: 0;
 		margin-inline-end: var(--mod-picker-spacing-text-to-icon, var(--spectrum-picker-spacing-text-to-icon));
 	}
 }
 
-.spectrum-Picker.is-open:not(.spectrum-Picker--quiet) {
-	color: var(--highcontrast-picker-content-color-default, var(--mod-picker-font-color-default-open, var(--spectrum-picker-font-color-default-open)));
-	background-color: var(--highcontrast-picker-background-color, var(--mod-picker-background-color-default-open, var(--spectrum-picker-background-color-default-open)));
-	border-color: var(--highcontrast-picker-border-color-default, var(--mod-picker-border-default-open, var(--spectrum-picker-border-color-default-open)));
-
-	&:hover {
-		color: var(--highcontrast-picker-content-color-default, var(--mod-picker-font-color-hover-open, var(--spectrum-picker-font-color-hover-open)));
-		background-color: var(--highcontrast-picker-background-color, var(--mod-picker-background-color-hover-open, var(--spectrum-picker-background-color-hover-open)));
-		border-color: var(--highcontrast-picker-border-color-hover, var(--mod-picker-border-color-hover-open, var(--spectrum-picker-border-color-hover-open)));
-
-		.spectrum-Picker-menuIcon {
-			color: var(--highcontrast-picker-content-color-default, var(--mod-picker-icon-color-hover-open, var(--spectrum-picker-icon-color-hover-open)));
-		}
+/* Disabled state. */
+.spectrum-Picker.spectrum-Picker:disabled,
+.spectrum-Picker.spectrum-Picker.is-disabled {
+	cursor: default;
+	background-color: var(--highcontrast-picker-background-color, var(--mod-picker-background-color-disabled, var(--spectrum-picker-background-color-disabled)));
+	border-color: var(--highcontrast-picker-border-color-disabled, transparent);
+	color: var(--highcontrast-picker-content-color-disabled, var(--mod-picker-font-color-disabled, var(--spectrum-picker-font-color-disabled)));
```

Comment:
```suggestion
.spectrum-Picker:disabled,
.spectrum-Picker.is-disabled {
	cursor: default !important;
	background-color: var(--highcontrast-picker-background-color, var(--mod-picker-background-color-disabled, var(--spectrum-picker-background-color-disabled))) !important;
	border-color: var(--highcontrast-picker-border-color-disabled, transparent) !important;
	color: var(--highcontrast-picker-content-color-disabled, var(--mod-picker-font-color-disabled, var(--spectrum-picker-font-color-disabled))) !important;
```

I think last time we spoke about doubling up on classes to increase specificity, there was concern this would not translate well to downstream consumers like web components who convert it to `:host`. I have read a lot lately in CSS that supports the use of !important when setting disabled styles to ensure they are always preferred and it best solves the web components use-case. Is there a reason why this might not work?

Reply:
I'm not keen on us having to start using `!important` for this for the typical reasons to avoid it (including it overriding inline styles and needing another !important to override it), though it may be more okay when the CSS is on the web components side.

I've used an alternative approach in the latest commit, that moves the disabled styles to the very end, to make sure it's after the hover media query that's causing the order issue. What do you think of that? Relying on the order isn't great either.

I wish this could be solved at the tooling level, so the CSS is not changed in unintended ways, but currently I do not see any configuration options available for `postcss-hover-media-feature`. I tried creating a separate `@media (hover: hover)` in the source CSS but it still bundled them all into one at the end.

Doubling up on the class felt like a lighter touch that just adds that little bit more specificity, without resulting to using`!important`. Another idea would be to double up the second part, `.spectrum-Picker:disabled:disabled`, if that's better with how web components converts it. Open to other ideas as well.